checker/src/org/checkerframework/checker/nullness/KeyForTransfer.java

```diff
@@ -11,21 +11,18 @@
 import org.checkerframework.dataflow.analysis.TransferResult;
 import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
 import org.checkerframework.dataflow.cfg.node.Node;
-import org.checkerframework.framework.flow.CFAnalysis;
-import org.checkerframework.framework.flow.CFStore;
-import org.checkerframework.framework.flow.CFTransfer;
-import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.flow.CFAbstractTransfer;
 import org.checkerframework.javacutil.AnnotationUtils;
 
 /*
  * KeyForTransfer ensures that java.util.Map.put and containsKey
  * cause the appropriate @KeyFor annotation to be added to the key.
  */
-public class KeyForTransfer extends CFTransfer {
+public class KeyForTransfer extends CFAbstractTransfer<KeyForValue, KeyForStore, KeyForTransfer> {
 
     protected final AnnotationMirror UNKNOWNKEYFOR, KEYFOR;
 
-    public KeyForTransfer(CFAnalysis analysis) {
+    public KeyForTransfer(KeyForAnalysis analysis) {
         super(analysis);
```

Comment:
I think subclasses only need to override the single argument constructor unless forcing forcing concurrent semantics is required.  I'll address this in a separate pull request.